src/common/qat.c

```diff
@@ -0,0 +1,537 @@
+}
+
+CpaStatus qat_hash_init(CpaInstanceHandle *cyInstHandle, 
+                        CpaCySymSessionCtx *sessionCtx,
+						CpaCySymHashAlgorithm hashAlg, 
+                        Cpa32U digestResultLenInBytes)
+{
+    CpaStatus status = CPA_STATUS_SUCCESS;
+    Cpa32U sessionCtxSize = 0;
+    CpaCySymSessionSetupData sessionSetupData = {0};
+
+    status = qaeMemInit();
+    if (CPA_STATUS_SUCCESS != status)
+    {
+        printf("Failed to initialize memory driver\n");
+        return 0;
+    }
+
+    status = icp_sal_userStartMultiProcess("SSL", CPA_FALSE);
+    if (CPA_STATUS_SUCCESS != status)
+    {
+        printf("Failed to start user process SSL\n");
+        qaeMemDestroy();
+        return 0;
+    }
+    
+    getInstance(cyInstHandle);
+    if (*cyInstHandle == NULL)
+    {
+        return CPA_STATUS_FAIL;
+    }
+
+    status = cpaCyStartInstance(*cyInstHandle);
+
+    if (CPA_STATUS_SUCCESS == status)
+    {
+        /*
+         * Set the address translation function for the instance
+         */
+        status = cpaCySetAddressTranslation(*cyInstHandle, virtToPhys);
+    }
+
+    if (CPA_STATUS_SUCCESS == status)
+    {
+        /*
+         * If the instance is polled start the polling thread. Note that
+         * how the polling is done is implementation-dependent.
+         */
+        startPolling(*cyInstHandle);
+
+        /*
+         * We now populate the fields of the session operational data and create
+         * the session.  Note that the size required to store a session is
+         * implementation-dependent, so we query the API first to determine how
+         * much memory to allocate, and then allocate that memory.
+         */
+        sessionSetupData.sessionPriority = CPA_CY_PRIORITY_NORMAL;
+        sessionSetupData.symOperation = CPA_CY_SYM_OP_HASH;
+        sessionSetupData.hashSetupData.hashAlgorithm = hashAlg;
+        sessionSetupData.hashSetupData.hashMode = CPA_CY_SYM_HASH_MODE_PLAIN;
+        sessionSetupData.hashSetupData.digestResultLenInBytes = digestResultLenInBytes;
+        sessionSetupData.digestIsAppended = CPA_FALSE;
+        sessionSetupData.verifyDigest = CPA_FALSE;
+
+        status = cpaCySymSessionCtxGetSize(
+            *cyInstHandle, &sessionSetupData, &sessionCtxSize);
+    }
+
+    if (CPA_STATUS_SUCCESS == status)
+    {
+        /* Allocate session context */
+        status = PHYS_CONTIG_ALLOC(sessionCtx, sessionCtxSize);
+    }
+
+    if (CPA_STATUS_SUCCESS == status)
+    {
+        status = cpaCySymInitSession(
+            *cyInstHandle, symCallback, &sessionSetupData, *sessionCtx);
+    }
+	printf("qat init status = %d\n", status);
+    return status;
+}
+
+CpaStatus qat_hash_update(CpaInstanceHandle *cyInstHandle, 
+                                    CpaCySymSessionCtx *sessionCtx, 
+                                    uint8_t *buf, 
+                                    size_t bufLen, 
+                                    uint8_t *csumBuf,
+									size_t csumLen,

```

Comment:
This line uses tabs while the other params use spaces. We use a size of 8 for our tabs.

Reply:
Fixed.